core/corehttp/gateway_handler.go

```diff
@@ -85,29 +93,93 @@ func (sw *statusResponseWriter) WriteHeader(code int) {
 	sw.ResponseWriter.WriteHeader(code)
 }
 
-func newGatewayHandler(c GatewayConfig, api coreiface.CoreAPI) *gatewayHandler {
-	unixfsGetMetric := prometheus.NewSummaryVec(
-		// TODO: deprecate and switch to content type agnostic metrics: https://github.com/ipfs/go-ipfs/issues/8441
+func newGatewaySummaryMetric(name string, help string) *prometheus.SummaryVec {
+	summaryMetric := prometheus.NewSummaryVec(
 		prometheus.SummaryOpts{
 			Namespace: "ipfs",
 			Subsystem: "http",
-			Name:      "unixfs_get_latency_seconds",
-			Help:      "The time till the first block is received when 'getting' a file from the gateway.",
+			Name:      name,
+			Help:      help,
+		},
+		[]string{"gateway"},
+	)
+	if err := prometheus.Register(summaryMetric); err != nil {
+		if are, ok := err.(prometheus.AlreadyRegisteredError); ok {
+			summaryMetric = are.ExistingCollector.(*prometheus.SummaryVec)
+		} else {
+			log.Errorf("failed to register ipfs_http_%s: %v", name, err)
+		}
+	}
+	return summaryMetric
+}
+
+func newGatewayHistogramMetric(name string, help string) *prometheus.HistogramVec {
+	// We can add buckets as a parameter in the future, but for now using static defaults
+	// suggested in https://github.com/ipfs/go-ipfs/issues/8441
+	defaultBuckets := []float64{0.1, 0.5, 1, 2, 3, 5, 8, 13}
```

Comment:
:question:

Reply:
@thibmeu [noted](https://github.com/ipfs/go-ipfs/issues/8441#issuecomment-1070528191):

> I feel like the granularity in the seconds unit is important, while there is very little information for blocks that are retrieved quickly.
> 
>I would suggest 50ms, 100ms, 250ms, 500ms, 1s, 2s, 5s, 10s. This is still arbitrary, but I feel it could capture more information.

Reply:
We could probably also use buckets for 30s, 60s (since read timeouts are set to 1min on the gateways)